### PR TITLE
[azure][feat] Add Azure Storage resource collection

### PR DESCRIPTION
**Pull request by 1101-1**
# Description

# To-Dos

- [x] I have created tests for any new or updated functionality.
- [x] I ran `tox` successfully.

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://inventory.fix.security/code-of-conduct).

### CODE REVIEW

**Comment by 1101-1** [comment]
Missing storage resources: 
https://learn.microsoft.com/en-us/rest/api/storagerp/blob-containers/list?view=rest-storagerp-2023-01-01&tabs=HTTP
and
https://learn.microsoft.com/en-us/rest/api/storagerp/blob-services/list?view=rest-storagerp-2023-01-01&tabs=HTTP

File: plugins/azure/fix_plugin_azure/resource/storage.py [hunk 1]
```diff
@@ -0,0 +1,1078 @@
...
+        "name": S("name"),
+        "table_signed_identifiers": S("properties", "signedIdentifiers")
+        >> ForallBend(AzureTableSignedIdentifier.mapping),
+        "table_name": S("properties", "tableName"),
+    }
+    table_signed_identifiers: Optional[List[AzureTableSignedIdentifier]] = field(default=None, metadata={'description': 'List of stored access policies specified on the table.'})  # fmt: skip
+    table_name: Optional[str] = field(default=None, metadata={"description": "Table name under the specified account"})
+
+
+resources: List[Type[AzureResource]] = [
```
Comment (aquamatthias) [inline plugins/azure/fix_plugin_azure/resource/storage.py:1073]: Please list all resource types here: AzureBlobContainer, etc. is missing
(even if it is collected via storage account)

File: plugins/azure/fix_plugin_azure/resource/storage.py [hunk 2]
```diff
@@ -0,0 +1,1078 @@
...
+    storage_account_sku_conversion_status: Optional[AzureStorageAccountSkuConversionStatus] = field(default=None, metadata={'description': 'This defines the sku conversion status object for asynchronous sku conversions.'})  # fmt: skip
+    supports_https_traffic_only: Optional[bool] = field(default=None, metadata={'description': 'Allows https traffic only to storage service if sets to true.'})  # fmt: skip
+
+    def _collect_items(
+        self,
+        graph_builder: GraphBuilder,
+        account_id: str,
+        service_type: str,
+        resource_type: str,
+        class_instance: AzureResource,
```
Comment (aquamatthias) [inline plugins/azure/fix_plugin_azure/resource/storage.py:937]: nifty - I like it.

File: plugins/azure/fix_plugin_azure/resource/storage.py [hunk 3]
```diff
@@ -0,0 +1,1078 @@
...
+        "migration_state": S("migrationState"),
+        "time_stamp": S("timeStamp"),
+    }
+    enabled: Optional[bool] = field(default=None, metadata={'description': 'This is an immutable property, when set to true it enables object level immutability at the container level.'})  # fmt: skip
+    migration_state: Optional[str] = field(default=None, metadata={'description': 'This property denotes the container level immutability to object level immutability migration state.'})  # fmt: skip
+    time_stamp: Optional[datetime] = field(default=None, metadata={'description': 'Returns the date and time the object level immutability was enabled.'})  # fmt: skip
+
+
+@define(eq=False, slots=False)
+class AzureBlobContainer(AzureResource):
```
Comment (aquamatthias) [inline plugins/azure/fix_plugin_azure/resource/storage.py:112]: Is a BaseBucket.

File: plugins/azure/fix_plugin_azure/resource/storage.py [hunk 4]
```diff
@@ -0,0 +1,1078 @@
...
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "access_policy": S("accessPolicy") >> Bend(AzureAccessPolicy.mapping),
+        "id": S("id"),
+    }
+    access_policy: Optional[AzureAccessPolicy] = field(default=None, metadata={"description": ""})
+    id: Optional[str] = field(default=None, metadata={'description': 'An unique identifier of the stored access policy.'})  # fmt: skip
+
+
+@define(eq=False, slots=False)
+class AzureFileShare(AzureResource):
```
Comment (aquamatthias) [inline plugins/azure/fix_plugin_azure/resource/storage.py:220]: Is a BaseNetworkShare

**Review by aquamatthias — APPROVED** [review]

**Pull request merged**